### DISCUSSION TITLE
My `runOnNoteTitleChange` does not work

### DISCUSSION THREAD

**Discussion by shirobachi** [TriliumNext/Trilium#3306] [Q&A] [answered]
Hello,

I want this script run when any note title changed. I am writing this note:
```
async function createNode(){
	const todoRootNote = await api.getNoteWithLabel('taskTodoRoot');
	const {note} = await api.createNote(todoRootNote.noteId, 'new task', '');
		
	return note.noteId;
}

createNode();
```
As a test. This note has label `#run=frontendStartup` and type JS backend: https://i.imgur.com/W6aiRK8.png
Other note has label `taskTodoRoot`: https://i.imgur.com/Jgrmyok.png

Manual running work (add new note)

--- 

I am working on a script for to-do tasks inspired on @nicollasricas great job. I am planning to generate script if note title has `[TODO]` so that why I need this.

Thanks in advance!
Simon

**Comment by nicollasricas** [comment]
Hey, If your script is running on frontend (`#run=frontendStartup`), the type should be frontend (`JS frontend`) as well also If you need to call functions from the backend API using the frontend environment, you need to encapsulate the backend code into the `api.runOnBackend`.

**Comment by zadam** [comment, marked as answer]
Hi, and where is your `runOnNoteTitleChange` defined? It should look like this (pointing to backend JS): 
![image](https://user-images.githubusercontent.com/617641/200935418-01032bd7-1d8a-474b-92f9-2dd97e600d56.png)
